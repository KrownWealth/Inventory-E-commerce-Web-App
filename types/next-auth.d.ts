import NextAuth from "next-auth";
import {UserType} from "@"

declare module "next-auth" {
  interface User {
    id: string;
    email: string;
    username: string;
    role: string; 
    image?: string | null;
    name?: string | null;
    accessToken?: string;
  }

  interface Session {
    user: User
  }

  interface JWT {
    role: string; 
    accessToken?: string;
  }
}
