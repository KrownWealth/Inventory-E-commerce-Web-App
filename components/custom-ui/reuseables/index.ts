import InputSearch from "./Search/searchBar";
import SideBarNav from "./Navigation/sideBar";
import UserProfile from "./Profile/userProfileDropdown";
import { BreadcrumbWithCustomSeparator } from "./Navigation/breadCrumb";
import ProductPageHead from "./Headings/productPageHead";
import ModalManager from "./Modals/handleModalOpenClose";
import PageHead from "./Headings/PageHeading";
import PriceFormulaModal from "./Modals/salesPriceFormula";
import CustomerTable from "./Admin/customersTable";
import OrderStatusBtn from "./Buttons/orderStatusBtn";
import OrdersTable from "./Admin/ordersTable";
import { DatePickerWithRange } from "./DateRange/dateRangePicker";
import UserAvatar from "./Profile/userAvatar";
import DashBoardCard from "./Cards/dashBoardCard";
import SalesBarCharts from "./Admin/salesBarCharts";
import SalesCategoryDoughnut from "./Admin/salesCategoryouhnutCharts";
import CustomerMap from "./Admin/customerMapLineChart";
import ProductTableTwo from "./Admin/productTableTwo";
import { ProductCardSkeleton } from "./Skeleton/productTableSkeleton";
import ProductsPagination from "./Pagination/productPagination";


import SignupWithGoogle from "./Authentication/google0Auth";

import LanguagSelect from "./Localization/languagesSelect";
import {Quote} from "./Authentication/quote";
import { LogoOnAuth } from "./Navigation/logoOnAuth";
import { WelcomeText } from "./Headings/welcomeText";

import { UserAvatarDropdown } from "./Navigation/avatarDropdown";
import { ProductCard } from "./Cards/ProductCard";
import UserSearch from "./Search/userSearchBar";
import AddCategoryModal from "./Modals/addCategoryModal";
import CategoryTable from "./Admin/categoryTable";



export { InputSearch, SideBarNav, UserProfile, BreadcrumbWithCustomSeparator, 
  ProductPageHead, ModalManager, PageHead, PriceFormulaModal, CustomerTable,
OrderStatusBtn, DatePickerWithRange, UserAvatar, DashBoardCard, SalesBarCharts, SalesCategoryDoughnut,
CustomerMap, ProductTableTwo, ProductCardSkeleton, ProductsPagination, OrdersTable,  LanguagSelect, CategoryTable,
SignupWithGoogle, Quote, LogoOnAuth, WelcomeText, UserAvatarDropdown, ProductCard, UserSearch, AddCategoryModal }