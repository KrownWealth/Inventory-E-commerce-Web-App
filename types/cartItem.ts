export interface CartItemType {
  productId: string; 
  userId: number;
  quantity: number;
  price: number;
  totalPrice?: number;
  image: string;
  name: string;
}
