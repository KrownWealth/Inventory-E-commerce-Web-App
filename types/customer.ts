export type CustomerType={
  customerId: string
  name: string
  email: string
  totalOrder: string
  totalSpent: string
}