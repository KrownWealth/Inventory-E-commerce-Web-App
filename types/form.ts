import { ChangeEvent } from "react";

export type FormFieldType = {
  label: string;
  htmlFor: string;
  type: string;
  id: string;
  isInvalid: boolean;
  errorMessage?: string;
  placeholder: string;
  startcnt?: React.ReactNode;
  endContent?: React.ReactNode;
  onChange: (value: string) => void; // Change event handler signature
  value: string;
  isRequired?: string;
  required?: boolean;
  height?: string;
};
