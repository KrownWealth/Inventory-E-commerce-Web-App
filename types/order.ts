export type OrderType={
  orderId: string
 productName: string
  departureDate: string
  deliveryDate: string
  totalAmount: string
  status: string
}

export type Order = {
  id: string;
  customerName: string;
  productName: string;
  orderDate: string;
  paymentStatus: string;
  totalPrice: number;
}