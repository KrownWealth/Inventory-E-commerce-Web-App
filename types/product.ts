export type ProductType = {
  id: string;
  image: string | null;
  name: string;
  sellingPrice: number | undefined; 
  stock: number;
  slug: string;
  category: {
    id: string;
    name: string;
  } | null;
  description: string;
  status: string;
  rating: string;
  review: string
};


export interface CategoryType {
    id: string;
    name: string;
}

export type ProductsType={
  id: string
  image: string | null; 
  name: string
  costPrice: number
  markupPercentage: number;
  sellingPrice: number | undefined;
  stock: number
  slug?: string
  category: {
        id: string;
        name: string;
    } | null;
  description: string
 status: string;
createdAt?: Date
updatedAt?: Date
}

