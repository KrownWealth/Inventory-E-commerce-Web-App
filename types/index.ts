import { ProductType, ProductsType } from "./product";
import { SalesProductType } from "./salesProduct";
import { CustomerType } from "./customer";
import { OrderType } from "./order";
import { FormFieldType, FormSchemaType } from "./form";

export type {ProductType, ProductsType, SalesProductType, CustomerType, OrderType, FormFieldType }