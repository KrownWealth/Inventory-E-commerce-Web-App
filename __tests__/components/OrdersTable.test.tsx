import { render, screen, fireEvent } from '@testing-library/react';
import { OrdersTable } from '@/components/custom-ui/reuseables';
import { Order } from '@/types';



const mockOrders: Order[] = [
  {
    id: '1',
    customerName: 'John Doe',
    productName: 'Laptop',
    orderDate: '2024-11-01',
    paymentStatus: 'Paid',
    totalPrice: 1200.00,
  },
  {
    id: '2',
    customerName: 'Jane Smith',
    productName: 'Smartphone',
    orderDate: '2024-11-02',
    paymentStatus: 'Pending',
    totalPrice: 800.00,
  },
];

describe('OrdersTable', () => {
  it('should render the OrdersTable with correct data', () => {
    render(<OrdersTable orders={mockOrders} />);


    expect(screen.getByText(/Order ID/i)).toBeInTheDocument();
    expect(screen.getByText(/Customer Name/i)).toBeInTheDocument();
    expect(screen.getByText(/Product Name/i)).toBeInTheDocument();
    expect(screen.getByText(/Order Date/i)).toBeInTheDocument();
    expect(screen.getByText(/Payment Status/i)).toBeInTheDocument();
    expect(screen.getByText(/Total Price/i)).toBeInTheDocument();
    expect(screen.getByText(/Actions/i)).toBeInTheDocument();


    expect(screen.getByText(/John Doe/i)).toBeInTheDocument();
    expect(screen.getByText(/Laptop/i)).toBeInTheDocument();
    expect(screen.getByText(/Paid/i)).toBeInTheDocument();
    expect(screen.getByText(/\$1,200.00/i)).toBeInTheDocument();

    expect(screen.getByText(/Jane Smith/i)).toBeInTheDocument();
    expect(screen.getByText(/Smartphone/i)).toBeInTheDocument();
    expect(screen.getByText(/Pending/i)).toBeInTheDocument();
    expect(screen.getByText(/\$800.00/i)).toBeInTheDocument();
  });

  it('should render with the correct structure', () => {
    const { container } = render(<OrdersTable orders={mockOrders} />);
    expect(container.firstChild).toHaveClass('border shadow-sm rounded-lg overflow-auto md:w-full w-[328px]');
    expect(container).toMatchSnapshot();
  });

  it('should open the dropdown menu and show actions', () => {
    render(<OrdersTable orders={mockOrders} />);

    const dropdownButton = screen.getAllByRole('button', { name: /Toggle menu/i })[0];
    fireEvent.click(dropdownButton);


    // expect(screen.getByText(/Accept/i)).toBeInTheDocument();
    // expect(screen.getByText(/Decline/i)).toBeInTheDocument();
  });
});
