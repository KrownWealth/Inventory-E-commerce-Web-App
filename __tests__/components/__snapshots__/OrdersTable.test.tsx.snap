// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OrdersTable > should render with the correct structure 1`] = `
<div>
  <div
    class="border shadow-sm rounded-lg overflow-auto md:w-full w-[328px]"
  >
    <div
      class="relative w-full overflow-auto"
    >
      <table
        class="w-full caption-bottom text-sm"
      >
        <thead
          class="[&_tr]:border-b"
        >
          <tr
            class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
          >
            <th
              class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
            >
              Order ID
            </th>
            <th
              class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
            >
              Customer Name
            </th>
            <th
              class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
            >
              Product Name
            </th>
            <th
              class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
            >
              Order Date
            </th>
            <th
              class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
            >
              Payment Status
            </th>
            <th
              class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
            >
              Total Price
            </th>
            <th
              class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
            >
              Actions
            </th>
          </tr>
        </thead>
        <tbody
          class="[&_tr:last-child]:border-0"
        >
          <tr
            class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
          >
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
            >
              1
            </td>
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr-0 font-medium"
            >
              John Doe
            </td>
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
            >
              Laptop
            </td>
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
            >
              2024-11-01
            </td>
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
            >
              Paid
            </td>
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
            >
              $1,200.00
            </td>
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
            >
              <button
                aria-expanded="false"
                aria-haspopup="true"
                class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-lg font-semibold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 w-10"
                data-state="closed"
                id="radix-:r4:"
                type="button"
              >
                <svg
                  class="h-4 w-4"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 256 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M214.059 377.941H168V134.059h46.059c21.382 0 32.09-25.851 16.971-40.971L144.971 7.029c-9.373-9.373-24.568-9.373-33.941 0L24.971 93.088c-15.119 15.119-4.411 40.971 16.971 40.971H88v243.882H41.941c-21.382 0-32.09 25.851-16.971 40.971l86.059 86.059c9.373 9.373 24.568 9.373 33.941 0l86.059-86.059c15.12-15.119 4.412-40.971-16.97-40.971z"
                  />
                </svg>
                <span
                  class="sr-only"
                >
                  Toggle menu
                </span>
              </button>
            </td>
          </tr>
          <tr
            class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
          >
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
            >
              2
            </td>
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr-0 font-medium"
            >
              Jane Smith
            </td>
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
            >
              Smartphone
            </td>
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
            >
              2024-11-02
            </td>
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
            >
              Pending
            </td>
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
            >
              $800.00
            </td>
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
            >
              <button
                aria-expanded="false"
                aria-haspopup="true"
                class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-lg font-semibold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 w-10"
                data-state="closed"
                id="radix-:r6:"
                type="button"
              >
                <svg
                  class="h-4 w-4"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 256 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M214.059 377.941H168V134.059h46.059c21.382 0 32.09-25.851 16.971-40.971L144.971 7.029c-9.373-9.373-24.568-9.373-33.941 0L24.971 93.088c-15.119 15.119-4.411 40.971 16.971 40.971H88v243.882H41.941c-21.382 0-32.09 25.851-16.971 40.971l86.059 86.059c9.373 9.373 24.568 9.373 33.941 0l86.059-86.059c15.12-15.119 4.412-40.971-16.97-40.971z"
                  />
                </svg>
                <span
                  class="sr-only"
                >
                  Toggle menu
                </span>
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
